lib/Message.php

```diff
@@ -53,6 +53,9 @@ function getBodyAsStream() {
             rewind($stream);
             return $stream;
         }
+        if (is_callable($body)) {
+            throw new \UnexpectedValueException('Callback to stream not supported');
```

Comment:
More than 'nice to have' actually ;) I think this needs to be an absolute requirement to not break existing stuff using the HTTP response.